scripts/Package_BGA/bga.py

```diff
@@ -50,40 +45,37 @@ def bga(args):
 
     padShape = Pad.SHAPE_CIRCLE
 
-    chamfer = min(1.0, min(pkgWidth, pkgHeight)*0.25)
-    silkOffset = 0.125
-    crtYd = 1.0
-
-    if pasteDiameter:
-        pasteRatio = (pasteDiameter / padDiameter - 1) / 2
-
+    chamfer = min(1.0, min(pkgWidth, pkgHeight) * 0.25)
+    silkOffset = 0.11
+    crtYdOffset = 1.0
+    
     def crtYdRound(x):
         # Round away from zero for proper courtyard calculation
         neg = x < 0
         if neg:
             x = -x
-        x = math.ceil(x * 100) / 100
+        x = math.ceil(x * 100) / 100.0
         if neg:
             x = -x
         return x
 
     xCenter = 0.0
-    xLeftFab = xCenter - pkgWidth / 2
-    xRightFab = xCenter + pkgWidth / 2
+    xLeftFab = xCenter - pkgWidth / 2.0
+    xRightFab = xCenter + pkgWidth / 2.0
     xChamferFab = xLeftFab + chamfer
-    xPadLeft = xCenter - pitch * ((layoutX - 1) / 2)
-    xPadRight = xCenter + pitch * ((layoutX - 1) / 2)
-    xLeftCrtYd = crtYdRound(xCenter - (pkgWidth / 2 + crtYd))
-    xRightCrtYd = crtYdRound(xCenter + (pkgWidth / 2 + crtYd))
+    xPadLeft = xCenter - pitch * ((layoutX - 1) / 2.0)
+    xPadRight = xCenter + pitch * ((layoutX - 1) / 2.0)
+    xLeftCrtYd = crtYdRound(xCenter - (pkgWidth / 2 + crtYdOffset))
+    xRightCrtYd = crtYdRound(xCenter + (pkgWidth / 2 + crtYdOffset))
```

Comment:
While you're forcing floating-point division elsewhere, you should probably update these lines too.